x/crosschain/keeper/keeper_cross_chain_tx_vote_inbound_tx.go

```diff
@@ -228,6 +228,7 @@ func (k msgServer) VoteOnObservedInboundTx(goCtx context.Context, msg *types.Msg
 		}
 		commit()
 		cctx.CctxStatus.ChangeStatus(types.CctxStatus_PendingOutbound, "")
+		k.RemoveInTxTrackerIfExists(ctx, cctx.InboundTxParams.SenderChainId, cctx.InboundTxParams.InboundTxObservedHash)
```

Comment:
Do we need to check this again if its already in the deferral?

Reply:
I'm not sure what you mean. This is just the cleanup for trackers which get finalized

Reply:
The defer function is already doing this.

Reply:
Probably it's a good catch.